src/Common/TargetSpecific.h

```diff
@@ -93,6 +93,13 @@ String toString(TargetArch arch);
 #define USE_MULTITARGET_CODE 1
 
 #if defined(__clang__)
+
+#define AVX512_FUNCTION_SPECIFIC_ATTRIBUTE __attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt,avx,avx2,avx512f")))
+#define AVX2_FUNCTION_SPECIFIC_ATTRIBUTE __attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt,avx,avx2")))
+#define AVX_FUNCTION_SPECIFIC_ATTRIBUTE __attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt,avx"))
+#define SSE42_FUNCTION_SPECIFIC_ATTRIBUTE __attribute__((target("sse,sse2,sse3,ssse3,sse4,popcnt")))
```

Comment:
How different is this from the default build with `ENABLE_SSE42` (which is on by default)?

I'm asking because it seems that, for dynamic dispatch, we are doing 3 builds: One for AVX2, one for SSE42 and the default one. If the default one and SSE42 are the same then we can save compilation and maybe also binary size by only leaving AVX2 and the default.

Reply:
@Algunenano this place was not optimized, for such scenario. If build is with SSE42 (by default) we generate additional SSE42 specialization (it can help if client build clickhouse on its own without SSE42). I do not expect that this will be a lot overhead, but we can detect such scenario in compile time using preprocessor and eliminate this.